src/saturn_engine/core/error.py

```diff
@@ -0,0 +1,12 @@
+import dataclasses
+
+
+@dataclasses.dataclass
+class ErrorMessageArgs:
+    exception_type: str
```

Comment:
All attribute are error related, no need to prefix them. I would also add `traceback` built from `TracebackData.from_exception(exception)`